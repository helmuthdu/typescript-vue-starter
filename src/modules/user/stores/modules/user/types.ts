export enum UserActionTypes {
  IS_LOGGED = 'USER/IS_LOGGED',
  SET_USER = 'USER/SET_USER',
  LOGIN = 'USER/LOGIN',
  LOGOUT = 'USER/LOGOUT'
}
