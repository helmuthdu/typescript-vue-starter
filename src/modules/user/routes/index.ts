import { USER_ROUTE_PATH, USER_ROUTE_NAME } from '@/modules/user/routes/paths';
import { RouteRecordRaw } from 'vue-router';

export const routes: Array<RouteRecordRaw> = [
  {
    path: USER_ROUTE_PATH.ROOT,
    component: () => import('../layouts/empty.layout.vue'),
    children: [
      {
        path: USER_ROUTE_PATH.SIGN_IN,
        name: USER_ROUTE_NAME.SIGN_IN,
        component: () => import('./sign-in/sign-in.route.vue')
      }
    ]
  }
];
