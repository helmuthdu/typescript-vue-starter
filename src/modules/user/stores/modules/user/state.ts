export interface State {
  username: string;
  email: string;
  isLoggedIn: boolean;
  token: string;
}

export const state = (): State => ({
  username: '',
  email: '',
  isLoggedIn: false,
  token: ''
});
