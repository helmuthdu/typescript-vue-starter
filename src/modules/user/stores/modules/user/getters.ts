import { State as AppState } from '../../';
import { GetterTree } from 'vuex';
import { State } from './state';
import { UserActionTypes } from './types';

export interface Getters<S, R> extends GetterTree<S, R> {
  [UserActionTypes.IS_LOGGED]: (state: S) => boolean;
}

export const getters: Getters<State, AppState> = {
  [UserActionTypes.IS_LOGGED](state) {
    return state.isLogged;
  }
};
