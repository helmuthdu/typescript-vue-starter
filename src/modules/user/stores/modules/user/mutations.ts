import { MutationTree } from 'vuex';
import { State } from './state';
import { UserActionTypes } from './types';

export const mutations: MutationTree<State> = {
  [UserActionTypes.SET_USER](state: State, payload: State) {
    state.username = payload.username || '';
    state.email = payload.email || '';
    state.isLogged = payload.isLogged || false;
    state.token = payload.token || '';
  }
};
