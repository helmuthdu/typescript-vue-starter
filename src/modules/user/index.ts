import { routes } from './routes';
import * as stores from './stores';

export type State = Readonly<{
  [stores.user.name]: stores.user.State;
}>;

export { routes, stores };
