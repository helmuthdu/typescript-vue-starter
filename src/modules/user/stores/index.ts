import * as user from './modules/user';

export type State = Readonly<{
  [user.name]: user.State;
}>;

export { user };
