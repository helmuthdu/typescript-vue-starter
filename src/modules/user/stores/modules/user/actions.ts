import { State as AppState } from '../../';
import { userApi, AuthenticatePayload } from '../../../api/user.api';
import { ActionContext, ActionTree } from 'vuex';
import { State } from './state';
import { UserActionTypes } from './types';

export interface Actions<S, R> extends ActionTree<S, R> {
  [UserActionTypes.LOGIN]: (context: ActionContext<S, R>, payload: any) => void;
  [UserActionTypes.LOGOUT]: (context: ActionContext<S, R>) => void;
}

export const actions: Actions<State, AppState> = {
  async [UserActionTypes.LOGIN]({ commit }, payload: AuthenticatePayload) {
    commit(UserActionTypes.SET_USER, {
      ...(await userApi.get(payload)).data,
      isLogged: true
    });
  },
  [UserActionTypes.LOGOUT]({ commit }) {
    commit(UserActionTypes.SET_USER, { username: '', email: '', isLogged: false });
  }
};
