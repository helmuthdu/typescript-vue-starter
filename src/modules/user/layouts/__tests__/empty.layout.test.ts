import { State } from '@/modules/user/stores/modules/user';
import { mount, VueWrapper } from '@vue/test-utils';
import { Store } from 'vuex';
import EmptyLayout from '../empty.layout.vue';

describe('User/Layout -> Empty', () => {
  let store: Store<State>;
  let wrapper: VueWrapper<any>;

  beforeEach(() => {
    wrapper = mount(EmptyLayout, { global: { plugins: [store] } });
  });

  it('should renders with props', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
