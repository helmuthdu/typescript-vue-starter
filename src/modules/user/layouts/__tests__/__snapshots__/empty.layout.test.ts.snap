// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth/Layout -> Empty should renders with props 1`] = `
<v-app-stub id="app">
  <v-content-stub tag="main" fill-height="">
    <router-view-stub name="default"></router-view-stub>
  </v-content-stub>
</v-app-stub>
`;
